#### Importing all the required libraries

In [0]:
import plotly.graph_objects as go
import pandas as pd
import plotly.tools

#### Reading the Nightingale dataset

In [0]:
df_nightingale = pd.read_excel("/content/drive/My Drive/Data Visualisation/nightingale-data.xlsx", header=1);

In [0]:
df_nightingale

Month  Average size of army  ...  Wounds & injuries.1  All other causes.1
0   Apr 1854                  8571  ...                  0.0                 7.0
1   May 1854                 23333  ...                  0.0                 4.6
2   Jun 1854                 28333  ...                  0.0                 2.5
3   Jul 1854                 28722  ...                  0.0                 9.6
4   Aug 1854                 30246  ...                  0.4                11.9
5   Sep 1854                 30290  ...                 32.1                27.7
6   Oct 1854                 30643  ...                 51.7                50.1
7   Nov 1854                 29736  ...                115.8                42.8
8   Dec 1854                 32779  ...                 41.7                48.0
9   Jan 1855                 32393  ...                 30.7               120.0
10  Feb 1855                 30919  ...                 16.3               140.1
11  Mar 1855                 30107  ...                 12.8                68.6
12  Apr 1855                 32252  ...                 17.9                21.2
13  May 1855                 35473  ...                 16.6                12.5
14  Jun 1855                 38863  ...                 64.5                 9.6
15  Jul 1855                 42647  ...                 37.7                 9.3
16  Aug 1855                 44614  ...                 44.1                 6.7
17  Sep 1855                 47751  ...                 69.4                 5.0
18  Oct 1855                 46852  ...                 13.6                 4.6
19  Nov 1855                 37853  ...                 10.5                10.1
20  Dec 1855                 43217  ...                  5.0                 7.8
21  Jan 1856                 44212  ...                  0.5                13.0
22  Feb 1856                 43485  ...                  0.0                 5.2
23  Mar 1856                 46140  ...                  0.0                 9.1

[24 rows x 8 columns]

#### Plotting BarPolar Chart based on Annual Mortality Rate from April 1855 to March 1856

In [0]:
fig= go.Figure()
fig.add_trace(go.Barpolar(
    r= df_nightingale["All other causes.1"][12:],
    theta = df_nightingale["Month"][12:],
    name="All other causes",
    marker=dict(color='rgb(0, 255, 255)', opacity=0.5)
))
fig.add_trace(go.Barpolar(
    r= df_nightingale["Wounds & injuries.1"][12:],
    theta = df_nightingale["Month"][12:],
    name='Wounds & injuries',
    marker=dict(color='rgb(241, 148, 138)', opacity=0.5)
))
fig.add_trace(go.Barpolar(
    r= df_nightingale["Zymotic diseases.1"][12:],
    theta = df_nightingale["Month"][12:],
    name= "Zymotic diseases",
    marker_color='rgb(144, 148, 151)' 
))
fig.update_layout(
    title='DIAGRAM of the CAUSES of MORTALITY in the ARMY in the EAST',
    font_size=14,
    legend_font_size=14,
    width=800,
    height=800,
    polar_angularaxis_direction= 'clockwise',
    polar=dict( radialaxis= dict(visible = False)),
    template ="plotly_white"
    )